### PR TITLE
Updates to build on OS X

### PR DESCRIPTION
**Pull request by boneill42**
I need some help here.. it's been about 15 years since I did C/POSIX.

I don't know the implications of NOT setting the process scheduler, but it appears as those OS X does not have the function:
https://www.gnu.org/software/gnulib/manual/html_node/sched_005fsetscheduler.html

For now, I used preprocessing to exclude this from the APPLE build.

### CODE REVIEW

File: tools/start-stop-daemon/start-stop-daemon.c [hunk 1]
```diff
@@ -39,6 +39,8 @@
 #  define OSFreeBSD
 #elif defined(__NetBSD__)
 #  define OSNetBSD
+#elif defined(__APPLE__)
+#  define OSLinux
```
Comment (richardkchapman) [inline tools/start-stop-daemon/start-stop-daemon.c]: I would have thought apple was closer to OSFreeBSD than OSLinux

**Comment by richardkchapman** [comment]
The effect will be that the -P option will not work:

       -P, --procsched policy:priority
              This alters the process scheduler policy and priority of the process before starting it. The priority can be optionally specified by appending a : followed by the value.
              The default priority is 0. The currently supported policy values are other, fifo and rr.

AFAIK we don't use that option in any of our usage of this program (though I'm not really very familiar with how we use it - @Michael-Gardner or @xwang2713 may know more).

**Comment by Michael-Gardner** [comment]
We don't use the -P option at all for start-stop-daemon.  It shouldn't be an issue if OS X doesn't support it.

File: tools/start-stop-daemon/start-stop-daemon.c (hunk 1 above)
Comment (boneill42) [inline tools/start-stop-daemon/start-stop-daemon.c]: I thought the same thing.  When using OSFreeBSD, I got the following incompatibilities:

In file included from /Users/bone/git/boneill42/hpcc/tools/start-stop-daemon/start-stop-daemon.c:65:
/usr/include/sys/proc.h:118:19: error: field has incomplete type 'struct itimerval'
        struct  itimerval p_realtimer;  /* Alarm timer. */
                          ^
/usr/include/sys/proc.h:118:9: note: forward declaration of 'struct itimerval'
        struct  itimerval p_realtimer;  /* Alarm timer. */
                ^
/Users/bone/git/boneill42/hpcc/tools/start-stop-daemon/start-stop-daemon.c:1263:9: warning: implicit declaration of function 'do_procinit' is invalid in C99 [-Wimplicit-function-declaration]
        do_procinit();

Comment (boneill42) [inline tools/start-stop-daemon/start-stop-daemon.c]: (which is why I punted and went with OSLinux)

File: tools/CMakeLists.txt [hunk 2]
```diff
@@ -39,3 +39,7 @@ if ( CMAKE_SYSTEM MATCHES Linux )
     message ("-- Building start-stop-daemon")
     HPCC_ADD_SUBDIRECTORY (start-stop-daemon)
 endif()
+
```
Comment (richardkchapman) [inline tools/CMakeLists.txt]: Would be preferable to say

```
  if ( CMAKE_SYSTEM MATCHES "Linux|APPLE" )
```
above (and at the same time update the comment that says start-stop-daemon is linux only)

Comment (boneill42) [inline tools/CMakeLists.txt]: Agreed, will do.

**Comment by boneill42** [comment]
Just a heads up, I'm pushing another change to this commit.  jfile.cpp had an issue on OS X, which resulted in a failed compile:

W20150402-123033 failed
<Result>
 <Exception><Code>4294967295</Code><Source>Roxie</Source><Message>UNIMPLEMENTED feature at /Users/bone/git/boneill42/hpcc/system/jlib/jfile.cpp(474)</Message></Exception>
</Result>

File: system/jlib/jfile.cpp [hunk 3]
```diff
@@ -470,8 +472,6 @@ bool CFile::setTime(const CDateTime * createTime, const CDateTime * modifiedTime
     if (modifiedTime)
         am.modtime  = timetFromIDateTime (modifiedTime);
     return (utime(filename, &am)==0);
-#else
```
Comment (boneill42) [inline system/jlib/jfile.cpp]: NOTE: This is where the issue was -- was failing the ECL compile because this was UNIMPLEMENTED for APPLE

**Comment by boneill42** [comment]
OK, with the final change to commonSetup.cmake, I am able to run esp, eclcc, dali and roxie on OS X.  I can issue the helloworld.ecl, and receive a valid result!

**Comment by boneill42** [comment]
Update: the plugins work as well, but I needed to copy the libraries into /opt/HPCCSystems/lib.

**Comment by richardkchapman** [comment]
Worth opening a JIRA for the plugins issue. I can take a look at that (or you can if you get to it first).

**Pull request merged**